src/operations/operation.ts

```diff
@@ -34,23 +34,42 @@ export interface OperationOptions extends BSONSerializeOptions {
  * a specific aspect.
  * @internal
  */
-export abstract class OperationBase<
-  T extends OperationOptions = OperationOptions,
-  TResult = Document
-> {
-  options: T;
+export abstract class OperationBase<TResult = Document> {
   ns!: MongoDBNamespace;
   cmd!: Document;
-  readPreference: ReadPreference;
+  readPreference!: ReadPreference;
   server!: Server;
+  session?: ClientSession;
   fullResponse?: boolean;
+  /** Explain returns the aggregation execution plan (requires mongodb 2.6 \>) */
+  explain?: boolean;
+
+  promoteBuffers?: boolean;
+  promoteValues?: boolean;
+  promoteLongs?: boolean;
+  raw?: boolean;
+  ignoreUndefined?: boolean;
 
   // BSON serialization options
-  bsonOptions?: BSONSerializeOptions;
+  get bsonOptions(): BSONSerializeOptions {
+    const bsonOptions: Document = {
+      promoteBuffers: this.promoteBuffers,
+      promoteValues: this.promoteValues,
+      promoteLongs: this.promoteLongs,
+      raw: this.raw,
+      ignoreUndefined: this.ignoreUndefined
+    } as const;
+
+    for (const key in bsonOptions) {
+      bsonOptions[key] === undefined && delete bsonOptions[key];
+    }
+
+    return bsonOptions;
+  }
 
-  constructor(options: T = {} as T) {
-    this.options = Object.assign({}, options);
-    this.readPreference = ReadPreference.primary;
+  constructor(options = {}) {
+    // this.readPreference = ReadPreference.primary;
```

Comment:
This looks like it needs to be cleaned up, was this a temporary measure or do we want to delete this line?

Reply:
Cleaned up the readPreference was previously "overwritten" if the options object specified it, but now it will be a property directly on the class.